src/shogun/kernel/GaussianARDKernel.cpp

```diff
@@ -65,22 +64,9 @@ CGaussianARDKernel* CGaussianARDKernel::obtain_from_generic(CKernel* kernel)
 
 float64_t CGaussianARDKernel::compute(int32_t idx_a, int32_t idx_b)
 {
-	REQUIRE(lhs && rhs, "Features not set!\n")
-
-	SGVector<float64_t> avec=
-			((CDenseFeatures<float64_t>*) lhs)->get_feature_vector(idx_a);
-	SGVector<float64_t> bvec=
-		((CDenseFeatures<float64_t>*) rhs)->get_feature_vector(idx_b);
-
-	REQUIRE(avec.vlen==bvec.vlen, "Number of right and left hand "
-			"features must be the same\n");
-
-	float64_t result=0;
-
-	for (index_t i = 0; i < avec.vlen; i++)
-		result += CMath::pow((avec[i]-bvec[i])*m_weights[i], 2);
+	float64_t result=distance(idx_a,idx_b);
```

Comment:
Can you please make sure we re-use the computational tricks from the Gaussian kernel?
Best would be to directly use the code.